scripts/merge-script/strategies/api-merge.ts

```diff
@@ -0,0 +1,191 @@
+    const cherryPickTargetBranches = targetBranches.filter(b => b !== githubTargetBranch);
+
+    // First cherry-pick the PR into all local target branches in dry-run mode. This is
+    // purely for testing so that we can figure out whether the PR can be cherry-picked
+    // into the other target branches. We don't want to merge the PR through the API, and
+    // then run into cherry-pick conflicts after the initial merge already completed.
+    const failure = await this._checkMergability(pullRequest, cherryPickTargetBranches);
+
+    // If the PR could not be cherry-picked into all  locally, then we know it can't be done
+    // through the Github API either. We abort merging and pass-through the failure.
+    if (failure !== null) {
+      return failure;
+    }
+
+    const mergeOptions: PullsMergeParams = {
+      pull_number: prNumber,
+      merge_method: method,
+      ...this.git.repoParams,
+    };
+
+    // If a commit message fixup is needed and we are merging using squash, we prompt
+    // for commit message changes. Note that this does not work with other merge methods
+    // as the Github API only allows commit message modifications for squash merging.
+    if (needsCommitMessageFixup && method === 'squash') {
+      await this._promptCommitMessageEdit(pullRequest, mergeOptions);
+    }
+
+    // Merge the pull request using the Github API into the selected base branch.
+    const {status, data: {sha}} = await this.git.api.pulls.merge(mergeOptions);
+
+    // https://developer.github.com/v3/pulls/#response-if-merge-cannot-be-performed
+    // Pull request cannot be merged due to merge conflicts.
+    if (status === 405) {
+      return {status: PullRequestStatus.MERGE_CONFLICTS, context: [githubTargetBranch]};
+    }
+    if (status !== 200) {
+      return {status: PullRequestStatus.UNKNOWN_MERGE_ERROR};
+    }
+
+    // If the PR does  not need to be merged into any other target branches,
+    // we exit here as we already completed the merge.
+    if (!cherryPickTargetBranches.length) {
+      return null;
+    }
+
+    // Refresh the target branch the PR has been merged into through the API. We need
+    // to re-fetch as otherwise we cannot cherry-pick the new commits into the remaining
+    // target branches.
+    this.fetchTargetBranches([githubTargetBranch]);
+
+    // Amount of commits that have landed in the target branch. This could vary from
+    // the amount of commits in the PR due to squashing.
+    const targetCommitsAmount = method === 'squash' ? 1 : pullRequest.commitsAmount;
+
+    // Cherry pick the merged commits into the remaining target branches.
+    const failedBranches = await this.cherryPickIntoTargetBranches(
+        `${sha}~${targetCommitsAmount}..${sha}`, cherryPickTargetBranches);
+
+    if (failedBranches.length) {
+      return {status: PullRequestStatus.MERGE_CONFLICTS, context: failedBranches};
+    }
+
+    this.pushTargetBranchesUpstream(cherryPickTargetBranches);
```

Comment:
Should we confirm the pushes are successful?  What happens if it were to fail here?

Reply:
If the push fails, then the script exits and the failure is printed out (git stderr). I don't think there is much we can do. This probably can happen if multiple PRs are merged at the same time. In those cases, the Git stderr message should explain this, and it's up to the consumer of this script to re-run (if possible).